merge.py

```diff
@@ -1,98 +1,303 @@
+            'pr_name': 'finish-pr-{}'.format(pr.get_num()),
+            'pr_num': pr.get_num(),
+        }
+        try:
+            cmds = [
+                {
+                    'cmd': 'git clone -b {branch} {repo_url} .',
+                    'desc': 'Clone',
+                    'error': 'Clone failed. Please try again.',
+                },
+                {
+                    'cmd': 'git remote add {remote_name} {apache_url}',
+                    'desc': 'Add Remote',
+                    'error': 'Failed to add remote. Please try again.',
+                },
+                {
+                    'cmd': 'git remote rename origin github',
+                    'desc': 'Rename Origin',
+                    'error': 'Failed to rename origin. Please try again.',
+                },
+                {
+                    'cmd': 'git config --local --add remote.github.fetch '
+                           '"+refs/pull/*/head:refs/remotes/github/pr/*"',
+                    'desc': 'Configure git fetch.',
+                    'error': 'Failed to configure git fetch. Please try again.',
+                    'shell': True,
+                },
+                {
+                    'cmd': 'git fetch --all',
+                    'desc': 'Fetch everything.',
+                    'error': 'Fetch failed. Please try again.',
+                },
+                {
+                    'cmd': 'git checkout -b {pr_name} github/pr/{pr_num}',
+                    'desc': 'Checkout PR',
+                    'error': 'Failed to check out PR. Please try again.',
+                },
+                {
+                    'cmd': 'git rebase {branch_path}',
+                    'desc': 'Rebase against target branch.',
+                    'error': 'Automatic rebase failed. Please manually rebase '
+                             'against {branch_path} and try again.',
+                },
+                {
+                    'cmd': 'git checkout {branch_path}',
+                    'desc': 'Check out target branch.',
+                    'error': 'Failed to check out {branch_path}. Please try '
+                             'again.',
+                },
+                {
+                    'cmd': 'git merge --no-ff -m "{msg}" {pr_name}',
+                    'desc': 'Merge PR',
+                    'error': 'Merge was not successful. Please try again.',
+                },
+                {
+                    'cmd': self.config['verification_command'],
+                    'desc': 'Verifying PR',
+                    'error': 'Verification failed. Please check the error log'
+                             ' and try again.',
+                },
+                # {
+                #    'cmd': 'git push {remote_name} HEAD:{branch}',
+                #    'desc': 'Push to remote master.',
+                #    'error': 'Remote push failed. Please try again.',
+                # },
+            ]
+            for command in cmds:
```

Comment:
I'd propose moving the for loop into the run_cmd method. So this becomes:

`self.run_commands(cmds, ...)`

That should simplify this call considerably. The tmp_dir can be setup in that function as well, so the logic to run a set of commands is entirely self contained.